src/main/java/communication/AbstractReceiver.java

```diff
@@ -121,30 +127,31 @@ public int getNumberOfReceivedMessages() {
 	protected void incrementNumberOfReceivedMessages() {
 		numberOfReceivedMessages++;
 	}
-	
+
 	/**
 	 * @return {@link #keygroupIDFilter}
 	 */
 	public KeygroupID getkeygroupIDFilter() {
 		return keygroupIDFilter;
 	}
-	
+
 	/**
-	 * Starts reception and interpretation of incoming envelopes..
+	 * Starts receiving and interpreting incoming envelopes..
 	 * 
-	 * @return <code>true</code> if reception was not running before and operation successful, 
-	 * <code>false</code> otherwise
+	 * @return a Future instance if reception was not running before and
+	 *         operation successful, <code>null</code> otherwise
 	 */
-	public Future<?> startReception() {
-		if (runnableFuture != null) {
-			if (!runnableFuture.isDone()) {
-				logger.error("Did not start reception because already running.");
-				return null;
-			}
+	public Future<?> startReceiving() {
+		if (runnableFuture != null && !runnableFuture.isDone()) {
```

Comment:
Is not this going to lead to a null pointer if runnableFuture is null? Or is the second condition not checked if the first is already false?